lightningd/plugin.c

```diff
@@ -930,6 +930,9 @@ static const char *plugin_opt_add(struct plugin *plugin, const char *buffer,
 
 	popt->name = tal_fmt(popt, "--%.*s", nametok->end - nametok->start,
 			     buffer + nametok->start);
+	/* an "|" alias could circumvent the unique-option-name check */
+	if (!deprecated_apis && tal_strreg(tmpctx, popt->name, "[|]"))
```

Comment:
I wonder why `strchr` isn't a better (more performant) choice here?

Reply:
Yes, I wasn't sure if we need to sanitize more, e.g. like "[^a-zA-Z0-9_-]" regex? Otherwise I'm happy to change it.

Currently setting an option like `~!@#$%^&*()+[{]};:=hello` in the config file just works.
I couldn't find any spec for valid `--longopts` chars.

Reply:
strchr here is fine. We assume plugins are not malicious :)